src/statsmodel.jl

```diff
@@ -45,12 +45,28 @@ struct DataFrameRegressionModel{M,T} <: RegressionModel
     mm::ModelMatrix{T}
 end
 
+"""
+    drop_intercept(::Type)
+
+Define whether a given model automatically drops the intercept. Return `false` by default. 
+To specify that a model type `T` drops the intercept, overload this function for the 
+corresponding type: `drop_intercept(::Type{T}) = true`
+
+Models that drop the intercept will be fitted without one: the intercept term will be 
+removed even if explicitly provided by the user. Categorical variables will be expanded 
+in the rank-reduced form (contrasts for `n` levels will only produce `n-1` columns).
+"""
+drop_intercept(::Type) = false
+
 for (modeltype, dfmodeltype) in ((:StatisticalModel, DataFrameStatisticalModel),
                                  (:RegressionModel, DataFrameRegressionModel))
     @eval begin
         function StatsBase.fit(::Type{T}, f::Formula, df::AbstractDataFrame,
                                args...; contrasts::Dict = Dict(), kwargs...) where T<:$modeltype
-            mf = ModelFrame(f, df, contrasts=contrasts)
+            trms = Terms(f)
+            drop_intercept(T) && (trms.intercept = true)
+            mf = ModelFrame(trms, df, contrasts=contrasts)
+            drop_intercept(T) && (mf.terms.intercept = false)
```

Comment:
Why do you need to call this again? If that's needed, would it be possible to group the two `drop_intercept(T)` calls in a single `if` block?

Reply:
I need the first call to  make sure that the categorical variable is expanded in the rank-reduced form (N-1 levels) and the second call to remove the intercept column. Otherwise this would fail if the user gave a categorical variable without intercept (I would get one level too many). There needs to be the ModelFrame call in between so I can't really make a single if block.

Reply:
OK, looks a bit weird, but well...

Reply:
you'll either duplicate the modelframe call or the drop_intercept call...

Reply:
Yeah, duplicating the `drop_intercept` call is better since it will be resolved statically.